.github/workflows/publish.yaml

```diff
@@ -2,8 +2,8 @@ name: Build and Publish Package to Pypi
 
 on:
   push:
-    branches:
-        - "main"
+    tags:
+      - "*"
 
 jobs:
   tests:
```

Comment:
I didn't find clean ways to create dependencies on tests. For now I copy pasted it here.

I can think of two other ways to do create a dependency to tests:
1. Move the publish section to the test workflow, and check the ref. If ref is tag, we run the publish job step. Otherwise we skip.
2. Use workflow calls. Create a callable workflow called tests. We call the callable test workflow in both tests and here before publish.

Reply:
> I didn't find clean ways to create dependencies on tests.

This sentence is not so easy to understand.. But okay.

> For now I copy pasted it here.

I am fine with that for now. We can improve our engineering practices later when we are more experienced. Let us focus more on the functional content of the project.